executor/builder.go

```diff
@@ -5196,9 +5196,25 @@ func (b *executorBuilder) buildCompactTable(v *plannercore.CompactTable) Executo
 		return nil
 	}
 
+	// check partition
+	if v.PartitionNames != nil {
+		if v.TableInfo.Partition == nil {
+			b.err = errors.Errorf("table:%s is not a partition table, but user specify partition name list:%+v", v.TableInfo.Name.O, v.PartitionNames)
+			return nil
+		}
+		for _, partitionName := range v.PartitionNames {
```

Comment:
Maybe better to avoid calling `FindPartitionDefinitionByName`, but just build a hash map and look up. This function's performance is low. Consider the case we have 1000 partitions, we will do 1000*1000 loops if we use it.